Patchwork/src/main/java/me/totalfreedom/display/Displayable.java

```diff
@@ -0,0 +1,442 @@
+package me.totalfreedom.display;
+
+import org.bukkit.Material;
+import org.bukkit.entity.HumanEntity;
+import org.bukkit.event.inventory.InventoryType;
+import org.bukkit.inventory.Inventory;
+import org.bukkit.inventory.InventoryHolder;
+import org.bukkit.inventory.ItemStack;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.ListIterator;
+
+public abstract class Displayable implements Inventory, InventoryHolder
+{
+
+    private final int size;
+    private final ItemStack[] contents;
+
+    protected Displayable(final int size)
+    {
+        if (size < 1 || size > 54)
+        {
+            throw new IllegalArgumentException("Invalid size for Displayable inventory");
+        }
+
+        this.size = (size % 9 == 0)
```

Comment:
Might want to replace this ternary operator with an if statement to improve code readability